packages/html-reporter/src/hideIfErrors.tsx

```diff
@@ -13,19 +13,20 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-import type { FullConfig, GlobalInfo } from '@playwright/test';
+import * as React from 'react';
 
-// We're dogfooding this, so the …/lib/… import is acceptable
-import * as ci from '@playwright/test/lib/ci';
+export class HideIfErrors extends React.Component<{}, { hitError: boolean }> {
+  state = {
```

Comment:
We don't use legacy components, only FCs w/ useState.

Reply:
Can you do React Error Boundaries with an FC? (The `componentDidCatch` is the special bit. The official docs only show class example.)

Reply:
> Only class components can be error boundaries.

(https://reactjs.org/docs/error-boundaries.html)

so we must use a class-based component here.

Reply:
Overall, throughout the code, we try avoiding silent exceptions. We also don't try/catch our own code, we prefer to handle the error where it happens instead. This is basically a silent exception in the React world, let's not use it.